# Atividade com Código 2 - FAP

## Instrução
Desenvolva um programa que recebe do usuário nome completo e ano de nascimento que seja entre 1922 e 2021. A partir dessas informações, o sistema mostrará o nome do usuário e a idade que completou, ou completará, no ano atual (2022).

Caso o usuário não digite um número ou apareça um inválido no campo do ano, o sistema informará o erro e continuará perguntando até que um valor correto seja preenchido.

## Resolução

A resolução foi feita em Pyhton 3 e depois implementada em JavaScript:

In [1]:
nome = input('Digite seu nome completo: ')

while True: # Checagem se o ano é inválido
  ano = input('\nDigite o seu ano de nascimento (1922-2021): ')
  try:
    ano = int(ano)
    if ((ano >= 1922) and (ano <= 2021)):
      break
    else:
      print('O ano informado está fora da faixa, digite novamente.')
      continue
  except:
    print('O ano informado é inválido, digite novamente.')
    continue

idade = 2022 - ano # Idade em 2022

print('\nSeu nome completo é: ', nome)
print(f'\nVocê completou {idade} anos em 2022.')

Digite seu nome completo: Pedro Artur Varela

Digite o seu ano de nascimento (1922-2021): 1921
O ano informado está fora da faixa, digite novamente.

Digite o seu ano de nascimento (1922-2021): 2022
O ano informado está fora da faixa, digite novamente.

Digite o seu ano de nascimento (1922-2021): 2002.2
O ano informado é inválido, digite novamente.

Digite o seu ano de nascimento (1922-2021): 2002

Seu nome completo é:  Pedro Artur Varela

Você completou 20 anos em 2022.


É visível que o código só saiu do loop quando um ano válido for colocado. Além disso, os erros de leitura foram corretamente sinalizados.

### Código Completo (JAVASCRIPT):

A única alteração feita, para o js, for substituir os _try_ e _except_ por _if_ e _else_, com o comando _!Number.isNaN(Number(ano))_ retornando verdadeiro se ano for um número válido. Além disso, o número foi tratado como um _parseFloat_ para que se o usuário digitar um float (Ex.: 2002.2) o código retornar que o valor é inválido, em decorrência de _Number.isInteger(ano) = false_ - não utilizei _parseInt_ porque ela converte os valores decimais em inteiros; se o usuário digitar um inteiro, mesmo com o _parseFloat_ esse valor vai ser entendido como um inteiro e, assim, _Number.isInteger(ano) = true_. 

```js
let ano;
let nome = prompt('Digite seu nome completo: ');

while (true) { // Checagem se o ano é inválido
  ano = prompt('\nDigite o seu ano de nascimento (1922-2021): ');
  if(!Number.isNaN(Number(ano))){
    ano = parseFloat(ano);
    if (ano >= 1922 && ano <= 2021 && Number.isInteger(ano)) {
      break;
    } else {
      console.log('O ano informado é inválido, digite novamente.');
      continue;
    }
  } else {
    console.log('O ano informado é inválido, digite novamente.');
    continue;
  }
}

let idade = 2022 - ano; // Idade em 2022

console.log('\nSeu nome completo é: ', nome);
console.log(`\nVocê completou ${idade} anos em 2022.`);
```